### PR TITLE
load full bitstream in flat design

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**
* load full bitstream in case for flat design
* added enable_flat option for xrt.ini
* it only support polling mode for now

### CODE REVIEW

**Review by mamin506 — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c [hunk 1]
```diff
@@ -120,7 +120,11 @@ zocl_load_bitstream(struct drm_zocl_dev *zdev, char *buffer, int length)
 		data[i+2] = temp;
 	}
 
-	return zocl_load_partial(zdev, data, bit_header.BitstreamLength);
+	/* On pr platofrm load partial bitstream and on Flat platform load full bitstream */
+	if (zdev->pr_isolation_addr)
+		return zocl_load_partial(zdev, data, bit_header.BitstreamLength);
+	else
+		return zocl_fpga_mgr_load(zdev, buffer, length, 0); //0 is for full btstream
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:127]: besides the typo, can we please change this to
else {
        /* 0 is for full bitstream */
        return ...
}

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c [hunk 2]
```diff
@@ -593,6 +597,17 @@ zocl_xclbin_read_axlf(struct drm_zocl_dev *zdev, struct drm_zocl_axlf *axlf_obj)
 		ret = zocl_load_aie_only_pdi(zdev, axlf, xclbin);
 		if (ret)
 			goto out0;
+	} else if ( axlf_obj->za_flags == DRM_ZOCL_PLATFORM_FLAT && 
+		    axlf_head.m_header.m_mode == XCLBIN_FLAT && 
+		    axlf_head.m_header.m_mode != XCLBIN_HW_EMU &&
+		    axlf_head.m_header.m_mode != XCLBIN_HW_EMU_PR) {
+		/* 
+		 * Load full bitstream in case, it is enables in xrt runtime config
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:605]: it is enabled?

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c [hunk 3]
```diff
@@ -593,6 +597,17 @@ zocl_xclbin_read_axlf(struct drm_zocl_dev *zdev, struct drm_zocl_axlf *axlf_obj)
 		ret = zocl_load_aie_only_pdi(zdev, axlf, xclbin);
 		if (ret)
 			goto out0;
+	} else if ( axlf_obj->za_flags == DRM_ZOCL_PLATFORM_FLAT &&
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:600]: you don't need space between "if (" and "axlf_obj."

**Review by larry9523 — CHANGES_REQUESTED** [review]
Looks good. Only some c-style nits. 
It is a good idea to run checkpacth to make sure no new warning is introduced.

**Comment by dayeh-xilinx** [comment]
retest this please

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c (hunk 2 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c]: not default enabled ... rephrased the comment

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c (hunk 3 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c]: fixed

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c (hunk 1 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c]: fixed

**Review by larry9523 — APPROVED** [review]

**Pull request merged**